### ISSUE TITLE
Add `change to file relative` and `change to project relative`

### ISSUE THREAD

**Issue by igorwojda** [vsch/idea-multimarkdown#662]
Structure
![image](https://user-images.githubusercontent.com/530122/49643663-3033fe00-fa0e-11e8-8f02-5b553c00d5cc.png)

Input `[Int queue](/app/src/test/java/com/igorwojda/datastructure/queue/int/IntQueue.md)`
`Change to file relative` -> `* [Int queue](../int/IntQueue.md)`

Input `* [Int queue](../int/IntQueue.md)`
`Change to file relative` -> `[Int queue](/app/src/test/java/com/igorwojda/datastructure/queue/int/IntQueue.md)`

**Comment by vsch** [comment]
@igorwojda, excellent suggestion. I had a hard time originally coming up with the labels. I keep looking at it as a GitHub repository centred instead of IntelliJ project.

`File` vs `page` is a no brainer. The issue comes up with `repo` vs `project` may be confusing because it really is relative to the VCS repository root, not project root directory. It may create some confusion when the project consists of multiple modules.

What do you think of making it `module` relative? It will be accurate for all project types.

**Comment by igorwojda** [comment]
Good point - this is the place where two worlds meet. Keep in mind that we are always using this plugin in the context of the local Idea project with local files. Honestly `page` was very confusing for me (page in this context has no obvious meaning since it's Idea IDE )

I suggest renaming 'repo' to 'Project root' (which is VCS root) and `page` to `file`

I am not sure how GitHub modules are structured and how this exactly relates to idea project, but purely from Idea project perspective this naming would make sense:
`Relative to project root` 
`Relative to this file` 
`Relative to module` 

and mabye 
`Relative to project VCS root` 
`Relative to module VCS root`

**Comment by vsch** [comment]
@igorwojda, originally the plugin was aimed at creating documentation which properly displays and links on GItHub. It has evolved beyond that now but link resolution still relies on GitHub link resolution rules to ensure the links in markdown files will work when viewed on GItHub.

There are no options beyond: file relative (not absolute), module relative (absolute ones starting with /, ie. no protocol prefix), http:// and file://, there are no other options, and short of adding a custom protocol would not be possible. Adding a custom protocol is out of the question because it would only be usable in the plugin and will make all links in the document useless outside the IDE.

There is no option to make the link relative to project because there is only GitHub link resolution engine. Ability to add multiple link resolution engines is on the TODO list but is a hard item because it affects code practically all over the plugin (since it did not have the concept of multiple engines).

When custom resolvers are added then the generated markdown will not have links which properly resolve on GitHub.

**Comment by igorwojda** [comment]
I didn't mean to add the custom protocol but thanks for the answer as now I have a better understanding of the problem. I guess the names are the only thing here that can be improved.

**Comment by vsch** [comment]
@igorwojda, don't worry about "not getting everything" in markdown limitations right away. Markdown's apparent simplicity is very misleading. It has a lot of caveats that are not obvious without spending a lot of time in the guts of its implementation.

No need to sensor your ideas on my account. Your suggestions will either let you learn quicker from my answers about what is there or why it cannot be, or I will pick up a great idea for the next "I can't work without it" feature.

Some features I really did not think of at all so they were a fresh look on how to work with markdown.

Others, were more of a conflict of interest. As developer and a user of the plugin I have a conflict of interest when it comes to adding new features. Yes they will save time for me as a user, but I have to spend a lot of time implementing them as a developer. So there is a natural tendency to avoid adding new features unless I am so fed up with the inefficiency that I can't take it.

When a user complains about something and I recognize that my only justification is "I don't want to put in the effort", I get motivated into seriously looking at the effort or the benefit of the result and usually wind up implementing it. 

So far without regrets about any of these since they led to other great features, as is always the case with good ideas.
